pinot-plugins/pinot-batch-ingestion/pinot-batch-ingestion-common/src/main/java/org/apache/pinot/plugin/ingestion/batch/common/SegmentGenerationTaskRunner.java

```diff
@@ -159,14 +161,24 @@ private SegmentNameGenerator getSegmentNameGenerator(SegmentGeneratorConfig segm
             Boolean.parseBoolean(segmentNameGeneratorConfigs.get(EXCLUDE_SEQUENCE_ID)),
             IngestionConfigUtils.getBatchSegmentIngestionType(tableConfig),
             IngestionConfigUtils.getBatchSegmentIngestionFrequency(tableConfig), dateTimeFormatSpec,
-            segmentNameGeneratorConfigs.get(SEGMENT_NAME_POSTFIX),
-            appendUUIDToSegmentName);
+            segmentNameGeneratorConfigs.get(SEGMENT_NAME_POSTFIX), appendUUIDToSegmentName);
       case BatchConfigProperties.SegmentNameGeneratorType.INPUT_FILE:
         String inputFileUri = _taskSpec.getCustomProperty(BatchConfigProperties.INPUT_DATA_FILE_URI_KEY);
         return new InputFileSegmentNameGenerator(segmentNameGeneratorConfigs.get(FILE_PATH_PATTERN),
-            segmentNameGeneratorConfigs.get(SEGMENT_NAME_TEMPLATE),
-            inputFileUri,
-            appendUUIDToSegmentName);
+            segmentNameGeneratorConfigs.get(SEGMENT_NAME_TEMPLATE), inputFileUri, appendUUIDToSegmentName);
+      case BatchConfigProperties.SegmentNameGeneratorType.UPLOADED_REALTIME:
+        Preconditions.checkState(segmentGeneratorConfig.getCreationTime() != null,
+            "Creation time must be set for uploaded realtime segment name generator");
+        Preconditions.checkState(segmentGeneratorConfig.getUploadedSegmentPartitionId() != -1,
+            "Valid partition id must be set for uploaded realtime segment name generator");
```

Comment:
can we also add sequence_id here?

Reply:
Do you mean for validation?

The sequenceId is sent as part of SegmentNameGenerator method interface

https://github.com/apache/pinot/blob/c2b1132c7e5f5b3e60155466debfb42ff3aa192d/pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/creator/name/SegmentNameGenerator.java#L40